src/anemoi/inference/condition.py

```diff
@@ -0,0 +1,285 @@
+import itertools
+import re
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Callable
+from typing import Self
+from typing import TypeVar
+
+import numpy as np
+import xarray as xr
```

Comment:
The xarray import is not lightweight (on Atos it takes about 30s), and since we don't use it by default, should we put the import where it is used (so in from_xarray) ?

Reply:
We should also not force people to install xarray to use `anemoi-inference` if they never need it. So, yes, we import in the method itself.

Reply:
Sure, I wasn't clear on thoughts around delayed imports as some linters really don't like it.
Fixed in upcoming commit